registration/register_server_led.go

```diff
@@ -2,164 +2,101 @@ package registration
 
 import (
 	"context"
-	"crypto/ed25519"
-	"crypto/x509"
-	"crypto/x509/pkix"
+	"crypto/hmac"
+	"crypto/sha256"
 	"fmt"
-	"math/big"
-	mathrand "math/rand"
 
 	"github.com/hashicorp/nodeenrollment"
 	"github.com/hashicorp/nodeenrollment/types"
-	"golang.org/x/crypto/curve25519"
-	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	"github.com/mr-tron/base58"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-// RegisterViaServerLedFlow registers a node, creating all keys and
-// certificates and returning the full set.
+// CreateServerLedActivationToken creates and stores a nonce and returns it;
+// this nonce can be used when a node requests to fetch credentials to authorize
+// it. The nonce is a serialized protobuf that also contains the creation time.
+// The serialized value is HMAC'd before storage.
 //
-// Note: there are currently no fields in the registration request but it is
-// required so that if fields are added it is not an API change.
+// The returned values are the activation token ID (used as the ID for storage)
+// and the token itself.
 //
-// Supported options: WithWrapper (passed through to LoadRootCertificates and
-// NodeInformation.Store), WithSkipStorage (useful for tests)
-func RegisterViaServerLedFlow(
+// Supported options: WithRandomReader, WithWrapper (passed through to
+// NodeInformation.Store), WithSkipStorage (useful for tests), WithState (to
+// encode state in the activation token)
```

Comment:
Can this comment be updated? It looks like WithSkipStorage is being used in Boundary for controller-led so documenting why that would be used outside of testing would be useful

Reply:
Took out the "useful for test" bit